### PR TITLE
Paginated method generation; not yet generating tests

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: src/Ast/AST.php [hunk 1]
```diff
@@ -534,11 +534,11 @@ public function toCode(): string
      *
      * @param Expression $condition The condition checked at the top of the while loop.
      *
-     * @return Callable The returned Callable returns an Expression once called with the loop code.
+     * @return Callable The returned Callable returns an while statemenet once called with the loop code.
```
Comment (jskeet) [inline src/Ast/AST.php:537]:
```suggestion
     * @return Callable The returned Callable returns a while statement once called with the loop code.
```

File: tests/ProtoTests/BasicPaginated/Gapic/BasicPaginatedGapicClient.php [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+ * ```
+ * $basicPaginatedServiceClient = new BasicPaginatedClient();
+ * try {
+ *     // Iterate over pages of elements
+ *     $pagedresponse = $basicPaginatedServiceClient->methodPaginated();
+ *     foreach ($pagedresponse->iteratePages() as $page) {
+ *         foreach ($page as $element) {
+ *             // doSomethingWith($element);
+ *         }
+ *
```
Comment (jskeet) [inline tests/ProtoTests/BasicPaginated/Gapic/BasicPaginatedGapicClient.php:53]: Can you remove this blank line easily?

File: tests/ProtoTests/BasicPaginated/Gapic/BasicPaginatedGapicClient.php [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+     * try {
+     *     // Iterate over pages of elements
+     *     $pagedresponse = $basicPaginatedServiceClient->methodPaginated();
+     *     foreach ($pagedresponse->iteratePages() as $page) {
+     *         foreach ($page as $element) {
+     *             // doSomethingWith($element);
+     *         }
+     *
+     *     }
+     * // Alternatively:
```
Comment (jskeet) [inline tests/ProtoTests/BasicPaginated/Gapic/BasicPaginatedGapicClient.php:190]: I assume this is the same whitespace issue that you already had?

**Review by jskeet — APPROVED** [review]
A few nitpicks, but generally fine.

File: tests/ProtoTests/BasicPaginated/Gapic/BasicPaginatedGapicClient.php (hunk 2 above)
Comment (chrisdunelm) [reply tests/ProtoTests/BasicPaginated/Gapic/BasicPaginatedGapicClient.php:53]: No, unfortunately it's not trivial to remove this.
I will revisit all the formatting issues in the generated code at a later date.

File: tests/ProtoTests/BasicPaginated/Gapic/BasicPaginatedGapicClient.php (hunk 3 above)
Comment (chrisdunelm) [reply tests/ProtoTests/BasicPaginated/Gapic/BasicPaginatedGapicClient.php:190]: Indeed yes. To be fixed later.

**Pull request merged**